src/Foundatio/Jobs/IQueueJob.cs

```diff
@@ -19,24 +19,37 @@ public interface IQueueJob<T> : IJob where T : class
 
 public static class QueueJobExtensions
 {
-    public static async Task RunUntilEmptyAsync<T>(this IQueueJob<T> job, TimeSpan acquireTimeout,
+    /// <summary>
+    /// Will run until the wait timeout expires. If there is still data the job will be cancelled. and then will cancel the job.
+    /// </summary>
+    /// <returns>The amount of queue items processed. This count will not be accurate if the job is cancelled.</returns>
+    public static async Task<int> RunUntilEmptyAsync<T>(this IQueueJob<T> job, TimeSpan waitTimeout,
         CancellationToken cancellationToken = default) where T : class
     {
-        if (acquireTimeout <= TimeSpan.Zero)
-            throw new ArgumentException("Acquire timeout must be greater than zero", nameof(acquireTimeout));
+        if (waitTimeout <= TimeSpan.Zero)
+            throw new ArgumentException("Acquire timeout must be greater than zero", nameof(waitTimeout));
 
         using var linkedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-        linkedCancellationTokenSource.CancelAfter(acquireTimeout);
+        linkedCancellationTokenSource.CancelAfter(waitTimeout);
 
         // NOTE: This has to be awaited otherwise the linkedCancellationTokenSource cancel timer will not fire.
-        await job.RunUntilEmptyAsync(linkedCancellationTokenSource.Token);
+        return await job.RunUntilEmptyAsync(linkedCancellationTokenSource.Token).AnyContext();
     }
 
-    public static Task RunUntilEmptyAsync<T>(this IQueueJob<T> job, CancellationToken cancellationToken = default) where T : class
+    /// <summary>
+    /// Will wait up to thirty seconds if queue is empty, otherwise will run until the queue is empty or cancelled.
+    /// </summary>
+    /// <returns>The amount of queue items processed. This count will not be accurate if the job is cancelled.</returns>
+    public static async Task<int> RunUntilEmptyAsync<T>(this IQueueJob<T> job, CancellationToken cancellationToken = default) where T : class
     {
         var logger = job.GetLogger();
-        return job.RunContinuousAsync(cancellationToken: cancellationToken, continuationCallback: async () =>
+
+        // NOTE: processed count is not accurate if the continuation callback is skipped due to cancellation.
+        int processed = 0;
+        await job.RunContinuousAsync(cancellationToken: cancellationToken, continuationCallback: async () =>
```

Comment:
this has an iteration count, but it will always be 1 for the queue job if cancelled immediately.